### ISSUE TITLE
Incorrect typings for GroupByFunction

### ISSUE THREAD

**Issue by gaopinghuang0** [omnidan/redux-undo#220]
The typings for `GroupByFunction` are as below.
```ts
  type GroupByActionTypes = (actions: string | string[]) => GroupByFunction;
  export type GroupByFunction = <State>(action: Action, currentState: State, previousHistory: StateWithHistory<State>) => any;
```
However, `GroupByActionTypes` is actually returning an arrow function accepting only one action.
```ts
export function groupByActionTypes (rawActions) {
  const actions = parseActions(rawActions)
  return (action) => actions.indexOf(action.type) >= 0 ? action.type : null
}
```
So I think the typing would be something like below.
```ts
  export type GroupByFunction = (action: Action) => any;
```

**Comment by louisrli** [comment]
+1: I'm also finding some misc typing issues here

**Comment by nmay231** [comment]
Resolved in #248 :+1:
